Automattic/wp-calypso
# A4A > Sites Dashboard: Fix the refresh issue with the favorite and monitor toggle functionality

Related to #

## Proposed Changes

![image](https://github.com/user-attachments/assets/576793ea-092d-4cef-aa09-c89579bc009a)


We had an issue when we clicked on the Favorite star and the Monitor to toggle the value. The Monitor toggle issue is in production, but the Favorite star fails with the PR that updates the DataViews package for the same issue.

The issue is with the `queryKey`, both actions (Favorite and Monitor switchers) works in an optimistic approach, that means that they update the React query cache without trying to refetch the sites. The problem is the way that the `queryKey` is created. For example, if the parameter is undefined and when the data was fetched the parameter is empty, the `queryKey` is different, so that is the reason that optimistic implementation fails because it doesn't find the site in the cache returned. The same happen if the `queryKey` creating doesn't follow the same parameter orders, if you introduce a parameter in a different order, the `queryKey` is different, and again, the optimistic implementation fails.

A part of this `queryKey` issue, the Monitor toggle had another issue, it had this missing parameter: `showOnlyDevelopmentSites`.

This PR fixes the `queryKey` issue, adding a simple helper function to create a hash of the parameter, regardless of the parameter order or if one of them is undefined or null. The hash function is simple; it sums the character codes of each parameter.

## Why are these changes being made?


*

## Testing Instructions

- Check the code
- Go to the `/sites` section
- Try to toggle the Monitor of one of your sites.
   - In trunk, it won't the value, but the query is successful. If you try to toggle again, an error will be displayed
   - This branch will refresh the value and work normally.
 - Check that the Favorite starts to continue working. It will be tested in [the DataViews updated branch](https://github.com/Automattic/wp-calypso/pull/93503).

## Pre-merge Checklist
